File: dsa_new/longest_sequence.py
class Solution:
    def longest(self,nums):
        num_set=set(nums)
        logest=0
        for num in nums:
            if num-1 not in num_set:
                length=1
                while num+length in num_set:
                    length+=1
                logest=max(logest,length)
        return logest
s=Solution()
nums=[1,2,3,4]
print(s.longest(nums))

File: dsa_new/longest_sequence.js
function longest(nums)
{
    let longest=0
    let num_set=new Set(nums)
    for(let num of nums)
    {
        if(!num_set.has(num-1))
        {
            let length=1
            while(num_set.has(num+length))
            {
                length+=1
            }
            longest=Math.max(longest,length)
        }
    }
    return longest
}
let nums=[1,2,3,4]
console.log(longest(nums))